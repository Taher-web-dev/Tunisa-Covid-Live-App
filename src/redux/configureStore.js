import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import updateReducer from './LATEST/reducer/reducer';

const reducer = combineReducers({
  latest: updateReducer,
});

const store = createStore(
  reducer,
  applyMiddleware(thunk, logger),
);
