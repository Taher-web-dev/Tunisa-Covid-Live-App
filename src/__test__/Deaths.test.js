import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import DeathsCases from '../components/Deaths/Deaths';

const mockStore = configureStore([]);

describe('render correctly DeathsCases component', () => {
  let store;
  let tree;
  beforeEach(() => {
    store = mockStore({
      latest: {},
      dayBefore: {},
      historical: {},
    });
    tree = renderer.create(
      <Provider store={store}>
        <DeathsCases />
      </Provider>,
    );
  });
  it('should render with given state from redux store', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
