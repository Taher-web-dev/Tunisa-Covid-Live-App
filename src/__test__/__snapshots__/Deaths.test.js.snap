// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly DeathsCases component should render with given state from redux store 1`] = `
<div
  className="confirmed-cases-page"
>
  <div
    style={
      Object {
        "height": "200px",
        "paddingBottom": "20px",
      }
    }
  >
    <canvas
      height={150}
      role="img"
      width={300}
    />
  </div>
  <p
    className="sub-title"
    style={
      Object {
        "backgroundColor": "rgb(226, 77, 120)",
        "fontSize": "18px",
        "margin": "0 ",
        "padding": "2.5% 0 2.5% 2.5%",
      }
    }
  >
    DEATHS BREAKDOWN
     
    2021-12-22
  </p>
  <div
    className="indicator-breakdown"
  >
    <div
      className="confirmed-case-item total-cases"
    >
      <p
        style={
          Object {
            "fontSize": "18px",
            "fontWeight": "700",
            "margin": "0 20% 0 5%",
            "width": "190px",
          }
        }
      >
        Total Deaths
      </p>
      <span
        style={
          Object {
            "fontSize": "18px",
            "fontWeight": "700",
            "marginRight": "8%",
            "width": "116px",
          }
        }
      >
        20
         
        cases
      </span>
      <img
        alt="back icone"
        src="fleche.png"
        style={
          Object {
            "marginRight": "2.5%",
          }
        }
      />
    </div>
    <div
      className=" confirmed-case-item  yesterday-total-cases"
    >
      <p
        style={
          Object {
            "fontSize": "18px",
            "fontWeight": "700",
            "margin": "0 20% 0 5%",
            "width": "190px",
          }
        }
      >
        Yesterday Total Deaths
      </p>
      <span
        style={
          Object {
            "fontSize": "18px",
            "fontWeight": "700",
            "marginRight": "8%",
            "width": "116px",
          }
        }
      >
        12
         
        cases
      </span>
      <img
        alt="back icone"
        src="fleche.png"
        style={
          Object {
            "marginRight": "2.5%",
          }
        }
      />
    </div>
    <div
      className=" confirmed-case-item today-vs-yesterday-confirmed"
    >
      <p
        style={
          Object {
            "fontSize": "18px",
            "fontWeight": "700",
            "margin": "0 20% 0 5%",
            "width": "190px",
          }
        }
      >
        Today Vs Yesterday New Deaths
      </p>
      <span
        style={
          Object {
            "fontSize": "18px",
            "fontWeight": "700",
            "marginRight": "8%",
            "width": "116px",
          }
        }
      >
        0.5
      </span>
      <img
        alt="back icone"
        src="fleche.png"
        style={
          Object {
            "marginRight": "2.5%",
          }
        }
      />
    </div>
    <div
      className=" confirmed-case-item growing-cases"
    >
      <p
        style={
          Object {
            "fontSize": "18px",
            "fontWeight": "700",
            "margin": "0 20% 0 5%",
            "width": "190px",
          }
        }
      >
        New Deaths Daily Growing Rate
      </p>
      <span
        style={
          Object {
            "fontSize": "18px",
            "fontWeight": "700",
            "marginRight": "8%",
            "width": "116px",
          }
        }
      >
        1.2
         
        %
      </span>
      <img
        alt="back icone"
        src="fleche.png"
        style={
          Object {
            "marginRight": "2.5%",
          }
        }
      />
    </div>
  </div>
</div>
`;
